django/urls/resolvers.py

```diff
@@ -256,7 +256,17 @@ def match(self, path):
         return None
 
     def check(self):
-        return self._check_pattern_startswith_slash()
+        warnings = self._check_pattern_startswith_slash()
+        if self._route.startswith('^') or self._route.endswith('$'):
```

Comment:
Perhaps adding a check for the route containing `(?P<` could be a good idea as well?

Reply:
Ooh, nice one...